### PR TITLE
[mypy] fix hashes folder

### PR DESCRIPTION
**Pull request by algobytewise**
### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: hashes/adler32.py [hunk 1]
```diff
@@ -15,15 +15,15 @@ def adler32(plain_text: str) -> str:
     Itterates and evaluates new value for each character
```
Comment (cclauss) [inline hashes/adler32.py:15]:
```suggestion
    Iterates and evaluates a new value for each character
```

File: hashes/adler32.py [hunk 2]
```diff
@@ -15,15 +15,15 @@ def adler32(plain_text: str) -> str:
...
+    '708642122'
     """
     MOD_ADLER = 65521
     a = 1
     b = 0
     for plain_chr in plain_text:
         a = (a + ord(plain_chr)) % MOD_ADLER
         b = (b + a) % MOD_ADLER
-    return (b << 16) | a
+    return str((b << 16) | a)
```
Comment (cclauss) [inline hashes/adler32.py:29]: Given that the example in https://en.wikipedia.org/wiki/Adler-32 returns an int, I would prefer that we do so as well.

File: hashes/enigma_machine.py [hunk 3]
```diff
@@ -41,8 +41,7 @@ def engine(input_character):
 
 
 if __name__ == "__main__":
-    decode = input("Type your message:\n")
-    decode = list(decode)
+    decode = list(input("Type your message:\n"))
```
Comment (cclauss) [inline hashes/enigma_machine.py:44]: Does this deal gracefully with the situation where the user enters the empty string?

File: hashes/sdbm.py [hunk 4]
```diff
@@ -26,12 +26,12 @@ def sdbm(plain_text: str) -> str:
...
 
     >>> sdbm('scramble bits')
-    730247649148944819640658295400555317318720608290373040936089
+    '730247649148944819640658295400555317318720608290373040936089'
     """
     hash = 0
     for plain_chr in plain_text:
         hash = ord(plain_chr) + (hash << 6) + (hash << 16) - hash
-    return hash
+    return str(hash)
```
Comment (cclauss) [inline hashes/sdbm.py:37]: Given that the sdbm code sited at https://en.wikipedia.org/wiki/List_of_hash_functions returns an int, I would prefer that we did so as well.

File: hashes/enigma_machine.py (hunk 3 above)
Comment (algobytewise) [reply hashes/enigma_machine.py:44]: In its current form, an empty input results in an empty code. Alternatives would be to throw an error or to ask the user again.

File: hashes/adler32.py [hunk 5]
```diff
@@ -9,21 +9,21 @@
 """
 
 
-def adler32(plain_text: str) -> str:
+def adler32(plain_text: str) -> int:
     """
     Function implements adler-32 hash.
-    Itterates and evaluates new value for each character
+    Itterates and evaluates a new value for each character
```
Comment (cclauss) [inline hashes/adler32.py:15]:
```suggestion
    Iterates and evaluates a new value for each character
```

**Review by cclauss — APPROVED** [review]

File: hashes/chaos_machine.py [hunk 6]
```diff
@@ -6,7 +6,8 @@
 m = 5
 
 # Buffer Space (with Parameters Space)
-buffer_space, params_space = [], []
+buffer_space: list = []
+params_space: list = []
```
Comment (dhruvmanila) [inline hashes/chaos_machine.py:10]: Please provide type hints for the list elements like so: `list[type]`

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

**Review by dhruvmanila — APPROVED** [review]

**Pull request merged**